nucypher/policy/conditions/address.py

```diff
@@ -0,0 +1,119 @@
+    """
+    A condition that checks if a user's wallet address is in a list of allowed addresses.
+    The user must provide a signed message to prove ownership of the wallet.
+    """
+
+    CONDITION_TYPE = ConditionType.ADDRESS_ALLOWLIST.value
+
+    class Schema(AccessControlCondition.Schema):
+        condition_type = fields.Str(
+            validate=validate.Equal(ConditionType.ADDRESS_ALLOWLIST.value),
+            required=True,
+        )
+
+        addresses = fields.List(
+            fields.String(required=True),
+            required=True,
+            validate=validate.Length(min=1, max=MAX_ALLOWLIST_ADDRESSES),
+        )
+
+        @validates("addresses")
+        def validate_addresses(self, addresses):
+            # Check for duplicates
+            if len(set(addresses)) != len(addresses):
+                raise ValidationError("Duplicate addresses are not allowed")
+
+            # Validate all addresses
+            for address in addresses:
+                try:
+                    # Ensure addresses have proper checksum
+                    normalized_address = to_checksum_address(address)
+                except ValueError as e:
+                    raise ValidationError(f"Invalid Ethereum address: {address}") from e
+                if normalized_address != address:
+                    raise ValidationError(
+                        f"Address {address} is not a checksummed address"
+                    )
+
+        @post_load
+        def make(self, data, **kwargs):
+            return AddressAllowlistCondition(**data)
+
+    def __init__(
+        self,
+        addresses: List[str],
+        name: Optional[str] = None,
+        condition_type: str = ConditionType.ADDRESS_ALLOWLIST.value,
+        *args,
+        **kwargs,
+    ):
+        """
+        Initialize a AddressAllowlistCondition.
+
+        Args:
+            addresses: List of checksummed Ethereum addresses that are allowed to decrypt
+            name: Optional name for the condition
+        """
+
+        # Store the validated addresses
+        self.addresses = addresses
+
+        super().__init__(
+            condition_type=condition_type,
+            name=name,
+            *args,
+            **kwargs,
+        )
+
+    def verify(self, **context) -> Tuple[bool, Union[None, Dict]]:
+        """
+        Verify if the user's wallet address is in the allowlist.
+
+        Args:
+            context: Dictionary containing context data, including the user's address
+
+        Returns:
+            Tuple containing:
+                - Boolean indicating if verification passed
+                - None (no additional data returned)
+        """
+        if not context:
+            raise InvalidConditionContext("No value provided for context variable")
+
+        # Get user's address using resolve_any_context_variables
+        user_address = resolve_any_context_variables(USER_ADDRESS_CONTEXT, **context)
```

Comment:
This is where the field I mentioned above would be used. Instead it would be something like

```suggestion
        user_address = resolve_any_context_variables(self.requester_address, **context)
```
(or whatever you end up calling the field)